src/cmd/manage_nsfs.js

```diff
@@ -360,57 +365,36 @@ function set_access_keys(argv, generate) {
 }
 
 // in name and new_name we allow type number, hence convert it to string
-async function fetch_account_data(argv, from_file) {
-    let data;
-    let access_keys = [{
-        access_key: undefined,
-        secret_key: undefined,
-    }];
-    let new_access_key;
-    const action = argv._[1] || '';
-    if (from_file) {
-        const fs_context = native_fs_utils.get_process_fs_context();
-        const raw_data = (await nb_native().fs.readFile(fs_context, from_file)).data;
-        data = JSON.parse(raw_data.toString());
-        // GAP - from-file is not validated
-    }
-    if (action !== ACTIONS.LIST && action !== ACTIONS.STATUS) _validate_access_keys(argv);
-    if (action === ACTIONS.ADD || action === ACTIONS.STATUS) {
-        const regenerate = action === ACTIONS.ADD;
-        access_keys = set_access_keys(argv, regenerate);
-    } else if (action === ACTIONS.UPDATE) {
-        access_keys = set_access_keys(argv, Boolean(argv.regenerate));
-        new_access_key = access_keys[0].access_key;
-        access_keys[0].access_key = undefined; //Setting it as undefined so we can replace the symlink
-    }
-
-    if (!data) {
-        data = _.omitBy({
-            name: _.isUndefined(argv.name) ? undefined : String(argv.name),
-            email: _.isUndefined(argv.name) ? undefined : String(argv.name), // temp, keep the email internally
-            creation_date: action === ACTIONS.ADD ? new Date().toISOString() : undefined,
-            wide: argv.wide,
-            new_name: _.isUndefined(argv.new_name) ? undefined : String(argv.new_name),
-            new_access_key,
-            access_keys,
-            nsfs_account_config: {
-                distinguished_name: argv.user,
-                uid: argv.user ? undefined : argv.uid,
-                gid: argv.user ? undefined : argv.gid,
-                new_buckets_path: argv.new_buckets_path,
-                fs_backend: argv.fs_backend ? String(argv.fs_backend) : config.NSFS_NC_STORAGE_BACKEND
-            }
-        }, _.isUndefined);
-    }
+async function fetch_account_data(action, user_input) {
+    const { access_keys, new_access_key } = get_access_keys(action, user_input);
+    let data = {
```

Comment:
don't we need the omitBy(..., _.isUndefined) ?

Reply:
Why do you think we need the omitBy(..., _.isUndefined)?
I would guess that someone would think we need this for the schema validation - but before use `JSON.parse` so the undefined are removed.

Reply:
just verifying with you that it's not needed anymore

Reply:
Not needed anymore, but if you think of a potential bug please add the scenario and I will test it.